src/cascadia/TerminalSettingsModel/Command.cpp

```diff
@@ -76,6 +77,11 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         return _subcommands ? _subcommands.Size() > 0 : false;
     }
 
+    bool Command::IsNestedCommand() const

```

Comment:
Wait I maybe missed the thread on why we need `IsNestedCommand` separate from `HasNestedCommands`. Looks like those are only different in the case where there's a command with nested commands that failed to parse. 

Is this because we have to get everything that _might_ have nested commands first, then parse the commands, then actually check if they successfully had nested commands?

Reply:
Yeah, I hate everything about this. We specifically need `IsNestedCommand` for the following case:
```js
{ "name": "foo", "commands": null }
```
In this very special case, `HasNestedCommands() == false` but `IsNestedCommand() == true`.

I'll add a function comment to clarify.